runtime/major_gc.c

```diff
@@ -896,7 +893,10 @@ static void mark_slice_darken(struct mark_stack* stk, value child,
   }
 }
 
-CAMLno_tsan /* Disable TSan reports from this function (see #11040) */
+CAMLno_tsan /* Loading from a location in the OCaml heap can cause false alarms
+               in TSan when this location is concurrently written to by
+               caml_modify. This false positive is due to the way we map OCaml
+               accesses to C11 accesses for TSan and is unlikely to go away. */
```

Comment:
This comment mentions an explanation, but it does not *explain* why the reports are false positives. This is better than what we had before, but from the comment I am still unable to understand what is going on. I would have a preference for an even longer comment (possibly at the place where the apparently-racy access is made?) that explains the race with `caml_modify` and why it should be considered a false positive.

Reply:
I will try to come up with a better comment in a new PR.

Reply:
See https://github.com/ocaml/ocaml/pull/12694.